swap/swap_test.go

```diff
@@ -682,6 +688,125 @@ func TestResetBalance(t *testing.T) {
 	}
 }
 
+// TestDebtCheques verifies that cheques that would put a node in debt past the defined tolerance are rejected
+// and that ones within the tolerance are accepted
+func TestDebtCheques(t *testing.T) {
+	testBackend := newTestBackend(t)
+	defer testBackend.Close()
+
+	creditorSwap, cClean := newTestSwap(t, beneficiaryKey, testBackend)
+	debitorSwap, dClean := newTestSwap(t, ownerKey, testBackend)
+	defer cClean()
+	defer dClean()
+
+	ctx := context.Background()
+	if err := testDeploy(ctx, creditorSwap, big.NewInt(0)); err != nil {
+		t.Fatal(err)
+	}
+	if err := testDeploy(ctx, debitorSwap, big.NewInt(int64(DefaultPaymentThreshold*2))); err != nil {
+		t.Fatal(err)
+	}
+
+	cDummyPeer := newDummyPeerWithSpec(Spec)
+	dDummyPeer := newDummyPeerWithSpec(Spec)
+	creditor, err := debitorSwap.addPeer(cDummyPeer.Peer, creditorSwap.owner.address, debitorSwap.GetParams().ContractAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+	debitor, err := creditorSwap.addPeer(dDummyPeer.Peer, debitorSwap.owner.address, debitorSwap.GetParams().ContractAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// the creditor and debitor should start out with a balance of 0
+	if creditor.getBalance() != 0 {
+		t.Fatalf("expected balance to be 0 for peer %v, but it is %d", creditor.ID(), creditor.getBalance())
+	}
+	if debitor.getBalance() != 0 {
+		t.Fatalf("expected balance to be 0 for peer %v, but it is %d", debitor.ID(), debitor.getBalance())
+	}
+
+	// no cheques should be present at this point
+	if creditor.getLastSentCheque() != nil {
+		t.Fatalf("expected no cheque sent to peer %v, but it is %v", creditor.ID(), creditor.getLastSentCheque())
+	}
+	if debitor.getLastReceivedCheque() != nil {
+		t.Fatalf("expected no cheque received from peer %v, but it is %v", debitor.ID(), debitor.getLastReceivedCheque())
+	}
+
+	// set asymmetric balance and attempt to send cheque
+	if err = creditor.setBalance(-int64(ChequeDebtTolerance * 2)); err != nil {
+		t.Fatal(err)
+	}
+	// now simulate sending the cheque to the creditor from the debitor
+	if err = creditor.sendCheque(); err != nil {
+		t.Fatal(err)
+	}
+	cheque := creditor.getPendingCheque()
```

Comment:
why do we need to create the cheque using `creditor`? wouldn't it be much simpler to just manually create the cheque (e.g. using `newSignedTestCheque`) and call `handleEmitChequeMsg` on `creditorSwap`? After all that's the only thing that really needs to be tested here.

Reply:
i honestly hadn't thought of this. i will reply down below.

Reply:
see reply [here](https://github.com/ethersphere/swarm/pull/1983#issuecomment-576332679)

Reply:
the reason why it's not used anywhere else (yet) is because the helper function is still new (merged only last week I think). There are already many other tests that are more complicated than they need to be and we shouldn't add new ones.

Reply:
got it. i will simplify this test further by calling this function as soon as #2078 is closed